<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtWidgetsApplication1Class</class>
 <widget class="QWidget" name="QtWidgetsApplication1Class">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1073</width>
    <height>835</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtWidgetsApplication1</string>
  </property>
  <widget class="QPushButton" name="OpenFileButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>20</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>載入圖片</string>
   </property>
  </widget>
  <widget class="QLabel" name="image1">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>10</y>
     <width>711</width>
     <height>621</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QGroupBox" name="GroupBox1">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>70</y>
     <width>331</width>
     <height>121</height>
    </rect>
   </property>
   <property name="title">
    <string>ROI</string>
   </property>
   <widget class="QPushButton" name="imageROI">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>影像分割</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="imageX2">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>70</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="imageY2">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>70</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="imageY1">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>30</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="imageX1">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>30</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'PMingLiU'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>10</y>
      <width>58</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>Y軸</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>10</y>
      <width>58</width>
      <height>15</height>
     </rect>
    </property>
    <property name="text">
     <string>X軸</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="histButton">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>20</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>影像直方圖</string>
   </property>
  </widget>
  <widget class="QSlider" name="ThresholdingSlider">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>220</y>
     <width>311</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>255</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>200</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="text">
    <string>影像二值化</string>
   </property>
  </widget>
  <widget class="QPushButton" name="EqualizationButton">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>250</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>直方圖等化</string>
   </property>
  </widget>
  <widget class="QPushButton" name="GrayButton">
   <property name="geometry">
    <rect>
     <x>120</x>
     <y>250</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>To Gray</string>
   </property>
  </widget>
  <widget class="QPushButton" name="HSVButton">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>250</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>To HSV</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="qtwidgetsapplication1.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>OpenFileButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>on_OpenFileButton_Clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>536</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>imageROI</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>imageROI()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>513</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>324</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>histButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>histimage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>33</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ThresholdingSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>Thresholding()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>EqualizationButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>HEqualization()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>66</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>HSVButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>ToHSV()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>GrayButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>ToGray()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>536</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_OpenFileButton_Clicked()</slot>
  <slot>imageROI()</slot>
  <slot>histimage()</slot>
  <slot>Thresholding()</slot>
  <slot>HEqualization()</slot>
  <slot>ToGray()</slot>
  <slot>ToHSV()</slot>
 </slots>
</ui>
